src/esw/cameras.py

```diff
@@ -1,387 +1,232 @@
-                # so we don't worry about this case.
-
-                # If same args, just create a new output
-                self.output_by_endpoint[endpoint] = jetson.utils.videoOutput(
-                    f"rtp://{endpoint}", argv=requested_resolution.arguments
-                )
-        else:
-            # If it does not exist already
-            try:
-                self.video_source = jetson.utils.videoSource(
-                    f"/dev/video{self.device}", argv=requested_resolution.arguments
-                )
-                self.video_source.Open()
-                self.output_by_endpoint[endpoint] = jetson.utils.videoOutput(
-                    f"rtp://{endpoint}", argv=requested_resolution.arguments
-                )
-                self.resolution = requested_resolution
-            except Exception:
-                rospy.logerr(f"Failed to create video source for device {self.device}.")
-                self.video_source = None
-                self.output_by_endpoint = {}
-
-    def is_streaming(self) -> bool:
-        """
-        Returns if streaming or not
-        :return: a bool if streaming
-        """
-        is_streaming = len(self.output_by_endpoint) != 0
-        if is_streaming and self.video_source is not None:
-            return True
-        elif not is_streaming and self.video_source is None:
-            return False
-        else:
-            assert False, "Should not have entered invalid state in is_streaming()."
+class StreamManager:
 
-
-class StreamingManager:
-    # _services is a list of size 2 representing the requested cameras from the primary and secondary laptops.
-    # For each laptop, there is a list of size 4 that represents information on the requested device and its resolution.
-    _services: List[List[CameraCmd]]
-
-    _device_lock: threading.Lock
-
-    # _list_of_resolution_options is a list of sizes that represents the different options for resolutions
-    # (0 for low, 2 for high).
-    # For each resolution, there is a Resolution object that has an option number and a list of strings that is passed
-    # into the jetson.utils function that is needed to create that specific resolution.
-    _list_of_resolution_options: List[Resolution]
-
-    # _endpoints is a list of size 8, where the first four strings represent the endpoints of the primary laptop
-    # and the last four strings represent the endpoints of the secondary laptop.
-    # Endpoints are just a combination of an IP and a port.
-    _endpoints: List[str]
-
-    # _service_streams_by_endpoints is a dictionary that maps each endpoint to the device that is being streamed to it.
-    _service_streams_by_endpoints: Dict[str, Tuple[int, int]]
-
-    # _video_devices is a list of jetson.utils objects. It can be None.
-    _video_devices: List[VideoDevices]
-    _active_devices: int
-
-    # determined by hardware. this is a constant. do not change.
-    MAX_CAMERA_DEVICES: int = 4
+    NUM_DIFFERENT_IPS = 2
```

Comment:
Why don't you just get the list of IPs from the yaml and find their length?

Reply:
Yup I could have :))